### PR TITLE
WIP: Add NodeJS cluster template

### PR DESCRIPTION
**Pull request by JohnNiang**
### What this PR does

As mentioned in title.

Currently, we have no related template icons, so I won't set icon first.

/cc @kubesphere-sigs/sig-devops

### CODE REVIEW

File: charts/ks-devops/templates/cluster-template-nodejs.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+apiVersion: devops.kubesphere.io/v1alpha3
+kind: ClusterTemplate
+metadata:
+  name: nodejs
+  annotations:
+    devops.kubesphere.io/displayNameEN: &defaultDisplayName "NodeJS"
+    devops.kubesphere.io/displayNameZH: "NodeJS"
+    devops.kubesphere.io/descriptionEN: &defaultDescription "Continuous integration for most NodeJS projects. Mainly include: dependency download, test, build and artifact archive."
```
Comment (JohnNiang) [inline charts/ks-devops/templates/cluster-template-nodejs.yaml:8]: Hi @Felixnoo, please help us review the description in English.

**Review by LinuxSuRen — COMMENTED** [review]
The feature looks good to me.

image

**Comment by JohnNiang** [comment]
/hold
I'm going to hold this PR due to failed verify: https://github.com/kubesphere-sigs/ks-devops-helm-chart/runs/5476016572?check_suite_focus=true

File: charts/ks-devops/templates/cluster-template-nodejs.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+      default: "node:14.19.0"
+      required: true
+  template: |
+    pipeline {
+      agent {
+        kubernetes {
+          inheritFrom 'nodejs base'
+          containerTemplate {
+            name 'nodejs'
+            image 'node:14.19.0'
```
Comment (JohnNiang) [inline charts/ks-devops/templates/cluster-template-nodejs.yaml:35]: Change image to `image: {{.params.NodeDockerImage}}`

File: charts/ks-devops/templates/cluster-template-nodejs.yaml (hunk 1 above)
Comment (Felixnoo) [inline charts/ks-devops/templates/cluster-template-nodejs.yaml:8]: `Continuous integration for most NodeJS projects. Mainly include: dependency download, test, build and artifact archive.`
->
`Designed for the continuous integration of most Node.js projects, including dependency downloading, testing, building, and artifact archiving.`

**Review by Felixnoo — COMMENTED** [review]
Please see my comment, thanks.

File: charts/ks-devops/templates/cluster-template-nodejs.yaml (hunk 1 above)
Comment (JohnNiang) [reply charts/ks-devops/templates/cluster-template-nodejs.yaml:8]: Thank you for reviewing! That's perfect!

**Comment by JohnNiang** [comment]
I'm going to close this PR due to https://github.com/kubesphere-sigs/pipeline-templates/pull/1 merged 
/close

**Pull request closed without merging**